checker/src/org/checkerframework/checker/lock/wholefiles/LockAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,261 @@
+        super(checker, useFlow);
+
+        LOCKHELD = AnnotationUtils.fromClass(elements, LockHeld.class);
+        LOCKPOSSIBLYHELD = AnnotationUtils.fromClass(elements, LockPossiblyHeld.class);
+        SIDEEFFECTFREE = AnnotationUtils.fromClass(elements, SideEffectFree.class);
+        GUARDEDBY = AnnotationUtils.fromClass(elements, GuardedBy.class);
+        JCIPGUARDEDBY = AnnotationUtils.fromClass(elements, net.jcip.annotations.GuardedBy.class);
+        JAVAXGUARDEDBY = AnnotationUtils.fromClass(elements, javax.annotation.concurrent.GuardedBy.class);
+        GUARDSATISFIED = AnnotationUtils.fromClass(elements, GuardSatisfied.class);
+
+        addAliasedAnnotation(javax.annotation.concurrent.GuardedBy.class, GUARDEDBY);
+        addAliasedAnnotation(net.jcip.annotations.GuardedBy.class, GUARDEDBY);
+
+        // This alias is only true for the Lock Checker. All other checkers must
+        // ignore the @LockingFree annotation.
+        addAliasedDeclAnnotation(LockingFree.class,
+                SideEffectFree.class,
+                AnnotationUtils.fromClass(elements, SideEffectFree.class));
+
+        // This alias is only true for the Lock Checker. All other checkers must
+        // ignore the @ReleasesNoLocks annotation.  Note that ReleasesNoLocks is
+        // not truly side-effect-free even as far as the Lock Checker is concerned,
+        // so there is additional handling of this annotation in the Lock Checker.
+        addAliasedDeclAnnotation(ReleasesNoLocks.class,
+                SideEffectFree.class,
+                AnnotationUtils.fromClass(elements, SideEffectFree.class));
+
+        postInit();
+    }
+
+    @Override
+    protected void postInit() {
+        super.postInit();
+
+        addInheritedAnnotation(AnnotationUtils.fromClass(elements,
+                MayReleaseLocks.class));
+        addInheritedAnnotation(AnnotationUtils.fromClass(elements,
+                ReleasesNoLocks.class));
+        addInheritedAnnotation(AnnotationUtils.fromClass(elements,
+                LockingFree.class));
+    }
+
+    @Override
+    public QualifierHierarchy createQualifierHierarchy(MultiGraphFactory factory) {
+        return new LockQualifierHierarchy(factory);
+    }
+
+    @Override
+    protected LockAnalysis createFlowAnalysis(List<Pair<VariableElement, CFValue>> fieldValues) {
+        return new LockAnalysis(checker, this, fieldValues);
+    }
+
+    @Override
+    public LockTransfer createFlowTransferFunction(CFAbstractAnalysis<CFValue, LockStore, LockTransfer> analysis) {
+        return new LockTransfer((LockAnalysis) analysis,(LockChecker)this.checker);
+    }
+
+    protected AnnotationMirror getDeclAnnotationNoAliases(Element elt,
+            Class<? extends Annotation> anno) {
+        String annoName = anno.getCanonicalName().intern();
+        return getDeclAnnotation(elt, annoName, false);
+    }
+
+    protected QualifierDefaults getQualifierDefaults() {
```

Comment:
Why aren't you using defaults directly?  It seems like most other checkers just use the field.